### PR TITLE
Add Category resource

### PR DESCRIPTION
**Pull request by rhannequin**
This feature adds Category resource:
* Get one category
* Get a list of categories
* Load category from a transaction
* Load category's parent from one category

### CODE REVIEW

File: lib/bankin/resources/category.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module Bankin
+  class Category < Resource
```
Comment (uncoder) [inline lib/bankin/resources/category.rb]: Nice job! :+1:

File: lib/bankin/resources/category.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    RESOURCE_PATH = '/v2/categories'
+
+    has_fields :id, :name
+
+    has_resource :parent, 'Category'
+
+    def self.list(options = {})
```
Comment (uncoder) [inline lib/bankin/resources/category.rb]: **.list** and **.get** methods are accessible for authorized bankin users and should accept **token** argument like **Account** resource. Also, it should be described in **User** resource as _auth_delegable_
```
auth_delegate :categories, class: 'Category', method: :list
auth_delegate :category, class: 'Category', method: :get
```
In this case you'll be able to call Bankin::Category methods on related resources like:
```ruby
bankin_user = Bankin::User.authenticate('email', 'password')

bankin_user.categories
bankin_user.category(id)
bankin_user.transactions.first.category
```

Comment (rhannequin) [reply lib/bankin/resources/category.rb]: Done in 60f4ee9.

**Comment by rhannequin** [comment]
Hello,
Do you need anything more for this PR?
Also will you merge https://github.com/quipuapp/bankin-api-ruby/pull/5, https://github.com/quipuapp/bankin-api-ruby/pull/4 and https://github.com/quipuapp/bankin-api-ruby/pull/2 ?

**Comment by rhannequin** [comment]
Hi,
Do you still maintain this repo? I am here to help.

**Pull request closed without merging**